protocols/tcp/tcp.go

```diff
@@ -114,8 +167,8 @@ func HandleTCP(ctx context.Context, conn net.Conn, md connection.Metadata, logge
 		}
 	}
 
-	// sending some random data
-	if err := server.sendRandom(conn); err != nil {
+	// Send port-specific or random response
+	if err := server.sendPortSpecificResponse(conn, md.TargetPort, logger); err != nil {
```

Comment:
These are service banners, you want to send them [before](https://github.com/mushorg/glutton/blob/5b68e3441834c86ff1ca9859498aac5632b64309/protocols/tcp/tcp.go#L154) reading from the connection.